addon/index.js

```diff
@@ -12,7 +12,64 @@
  * ```
  */
 export function proto(target, key, desc) {
-  desc.value = desc.initializer();
+  desc.value = desc.initializer.apply(target);
```

Comment:
Probably better to use `.call` here since we don't need to pass an array of args...